app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/add/AddHealthRecordsFragment.kt

```diff
@@ -73,7 +99,31 @@ class AddHealthRecordsFragment : Fragment(R.layout.fragment_health_records) {
         binding.toolbar.ivRightOption.apply {
             visibility = View.VISIBLE
             setOnClickListener {
-                findNavController().navigate(R.id.settingFragment)
+                findNavController().navigate(R.id.profileFragment)
+            }
+        }
+    }
+
+    private fun checkLogin() {
+        bcscAuthViewModel.checkLogin()
+
+        viewLifecycleOwner.lifecycleScope.launch {
```

Comment:
do not put this flow observer in the method put it directly in the onViewCreate

Reply:
Observer will emit login = true/false as soon as it is observed. This will break the screen flow is observed immediately .